doc/flatpak-metadata.xml

```diff
@@ -1094,6 +1094,33 @@
             <!-- TODO: More information would be nice -->
           </para>
         </refsect2>
+        <refsect2 id="usb-devices">
+          <title>[USB Devices]</title>
+          <para>
+            USB devices can be allowed or blocked by the USB portal as specified
+            by the keys in this group. The vendor and product ids are validated
+            by Flatpak, but aren't otherwise used or parsed.
+            Available since 1.15.7.
+          </para>
+          <variablelist>
+                <varlistentry>
+                    <term><option>allowed-devices</option> (string list)</term>
+                    <listitem><para>
+                        List of allowed USB devices. Each element of can be composed
+                        of "VENDOR_ID:PRODUCT_ID", or "VENDOR_ID:*". Available since
```

Comment:
I left that out somewhat on purpose, because I wanted to think better about how to handle these different queries through the command line. So far there is:

 * Specific device: `--usb=AAAA:BBBB`
 * All devices from specific vendor: `--usb=AAAA:*`

For class and subclass queries, I can imagine the following options:

 1. `--usb=class=AA:[BB|*]`
 2. `--usb-class=AA:[BB|*]`
 3. `--usb=AA:[BB|*]`

I personally prefer (1) and (2). However, does that mean devices and vendor queries need to be adjusted into `--usb=device=AAAA:BBBB`, and `--usb=vendor=AAAA`? It looks a bit convoluted to me.

An option for allowlisting all devices is `--usb=*:*`. Or a contracted form, `--usb=*`.

Thoughts?

Reply:
I think maybe the usb selector could be made more generic.
Say, a per element prefix of dev/cls/vnd, and maybe allow combinations.
So `cls:AA:*+vnd=BB`. We could have a generic --usb argument taking these, or helper --usb-device=... that constructs the generic key.

Reply:
Thanks for the suggestion @alexlarsson, I'll try and implement that

Reply:
With this last push, I've applied this `cls`/`vnd`/`dev` syntax to `--usb` and `--no-usb`. Some rules apply here:

 * `--usb=all` must not have any further rules (e.g. `--usb=all+vnd:0fd9` is in valid). There are tests for that.
 * `dev`  rules must have a `vnd` rule together, but `vnd` is okay to not have a matching `dev`. e.g. `--usb=vnd:0fd9` is valid, `--usb=vnd:0fd9+dev:0060` is valid, but --usb=dev:0060` is invalid. There are tests for that.
 * `cls` rules can be freely applied on top of `dev` and `vnd`. There are tests for that.

Notice that I haven't yet added the helper `--usb-device`. If people seem satisfied with this approach, I'll be happy to add the helper :slightly_smiling_face: 

Please let me know what you think!

Reply:
As I mentionnd in the other thread https://github.com/flatpak/flatpak/pull/5620#discussion_r1518719742

I can see issues about using `*` in any of the command line arguments.